pkg/client/client.go

```diff
@@ -396,6 +396,9 @@ func (c *Client) request(ctx context.Context, method string, path string, body i
 	req, err := http.NewRequest(method, "http://"+c.endpoint+path, body)
 	req = req.WithContext(ctx)
 
+	// add auth token
+	req.Header.Add("X-Token", c.cfg.Client.Token)
```

Comment:
BTW, I'd recommend using the standard `Authorization: Bearer` convention: https://swagger.io/docs/specification/authentication/bearer-authentication/.